django_celery_monitor/admin.py

```diff
@@ -191,6 +194,25 @@ def kill_tasks(self, request, queryset):
                 revoke(state.task_id, connection=connection,
                        terminate=True, signal='KILL')
 
+    @action(_('Retry selected tasks'))
+    def retry_tasks(self, request, queryset):
+        with current_app.default_connection() as connection:
+            for state in queryset:
+                module_name, task_name = state.name.rsplit('.', 1)
+                task = getattr(import_module(module_name), task_name, None)
+                if task:
+                    kwargs = {
+                        'args': ast.literal_eval(state.args),
+                        'kwargs': ast.literal_eval(state.kwargs),
+                        'countdown': RETRY_TASK_COUNTDOWN,
+                        'connection': connection
+                    }
+                    try:
+                        task.apply_async(**kwargs)
+                    except TypeError:
+                        # it must be a class task
+                        task().apply_async(**kwargs)
```

Comment:
Hmm, class based tasks are essentially dead in Celery 4.x, so it's better to have this raise a TypeError if something goes on during the call of apply_async.